app/sider/Dokumentasjon.tsx

```diff
@@ -28,40 +30,70 @@ export default function DokumentasjonSide() {
   const teksterDokumentasjon = useTekster(ESanityMappe.DOKUMENTASJON);
   const teksterFelles = useTekster(ESanityMappe.FELLES);
   const teksterSendEndringer = useTekster(ESanityMappe.SEND_ENDRINGER);
-  const [endringsmelding] = useEndringsmelding();
+  const [endringsmelding, settEndringsmelding] = useEndringsmelding();
   const [, settEndringsmeldingMottattDato] = useEndringsmeldingMottattDato();
 
   const navigate = useNavigate();
   const submit = useSubmit();
 
   const action = hentAction(ytelse);
-  const actionData: IPostResponse | undefined = useActionData<typeof action>();
+  const actionData: IPostResponse | IPostFilResponse | undefined =
+    useActionData<typeof action>();
 
   const [feilKode, settFeilKode] = useState<EFritekstFeil | null>(null);
 
   useEffect(() => {
     if (!actionData) return;
-
-    if (actionData.status === EStatusKode.OK && actionData.data) {
+    if (
+      actionData.status === EStatusKode.OK &&
+      'data' in actionData &&
+      actionData.data
+    ) {
       settEndringsmeldingMottattDato(actionData.data.mottattDato);
       navigate(hentPathForSteg(ytelse, ESteg.KVITTERING));
-    } else {
+    } else if (
+      actionData.status === EStatusKode.OK &&
+      'response' in actionData &&
+      actionData.response
+    ) {
+      actionData as IPostFilResponse;
+      const fil: IFil = actionData.response;
+      const dokumenter = endringsmelding.dokumenter;
+      dokumenter.push(fil.dokumentId);
+      const nyEndringsmelding: IEndringsmelding = {
+        dokumenter: dokumenter,
+        tekst: endringsmelding.tekst,
+      };
+      settEndringsmelding(nyEndringsmelding);
+    } else if ('data' in actionData) {
       settFeilKode(actionData.feilKode || null);
     }
-  }, [actionData, navigate, settEndringsmeldingMottattDato, ytelse]);
+  }, [
+    actionData,
+    navigate,
+    settEndringsmeldingMottattDato,
+    ytelse,
+    settEndringsmelding,
+    endringsmelding.dokumenter,
+    endringsmelding.tekst,
+  ]);
 
   function håndterSendEndringsmelding() {
+    let dokumenterStreng = '';
+    endringsmelding.dokumenter.forEach(dok => (dokumenterStreng += dok + ','));
```

Comment:
Dette bør sikkert gjøres på en annen måte som nevnt i denne [PRen](https://github.com/bekk/nav-familie-endringsmelding-api/pull/15) der man sender med hele endringsmeldingen og ikke to strenger osv.